Add NodeJS Year Service with OTLP

JamieDanielson opened the pull request:
Add NodeJS Year Service with OTLP
Adds support for environment variables

Side note: `//prettier-ignore`s added to keep imports from wrapping to multi-line with printWidth of 80.

Closes #44

vreynolds commented on Tiltfile:
```diff
@@ -148,13 +148,39 @@ def launch_dotnet_name_service(auto_init=True):
+    env = {
+        'SERVICE_NAME': name,
+        'HONEYCOMB_API_KEY': apikey,
+        'HONEYCOMB_DATASET': dataset,
```
as a dogfood resident, may I request `HONEYCOMB_API` (for endpoint) as well?

vreynolds commented on node/docker-compose.yml:
```diff
@@ -1,12 +1,34 @@
 services:
+  collector:
```
don't need the collector

vreynolds commented on node/year-service/main.js:
```diff
@@ -0,0 +1,69 @@
+metadata.set('x-honeycomb-team', HONEYCOMB_API_KEY);
+metadata.set('x-honeycomb-dataset', HONEYCOMB_DATASET);
+
+provider.addSpanProcessor(
```
initialize tracing separately and include with `node -r`

vreynolds reviewed (changes requested):
comment

JamieDanielson commented:
* Split Open Telemetry to separate tracing file in Year service
* Added/updated environment variables, including OTLP_ENDPOINT
* Remove collector (not used in implementation)
* Added README for Node Services

vreynolds commented on Tiltfile:
```diff
@@ -148,13 +148,43 @@ def launch_dotnet_name_service(auto_init=True):
+    flags: (optional) any additional flags to add to the command line
+    '''
+    
+    apikey = os.getenv('HONEYCOMB_API_KEY')
```
I don't think we need to do this mapping, the vars will be available to the process without `serve_env` (we only need to do that if we want to define variable in the Tilt file itself)

vreynolds replied:
Oh I think we would only keep the service name one, since that's defined through tilt

JamieDanielson replied:
Makes sense - good call! I removed those extra variable declarations and confirmed `tilt up` still works based on env vars. Kept Service Name.

vreynolds reviewed (approved):
💥
